### PR TITLE
chore: migrate NftBalanceQuery

### PR DESCRIPTION
**Pull request by cbachmeier**
Migrated NftBalance Query to Apollo. This required refactoring the WalletAsset type to have all params potentially be unidentified. I also made a change to the Codegen config so that egenerated types aren't wrapped in `Maybe` as opposed to just having value vs being undefined.

### CODE REVIEW

File: apollo-codegen.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ const config: CodegenConfig = {
       plugins: ['typescript', 'typescript-operations', 'typescript-react-apollo'],
       config: {
         withHooks: true,
+        // This avoid all generated schemas being wrapped in Maybe https://the-guild.dev/graphql/codegen/plugins/typescript/typescript#maybevalue-string-default-value-t--null
```
Comment (justindomingue) [inline apollo-codegen.ts:14]: is it safe though?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 2]
```diff
@@ -214,10 +214,14 @@ const PriceTextInput = ({
       >
         {focused ? (
           <>
-            <Row display={asset.lastPrice ? 'flex' : 'none'} marginRight="8">
-              LAST: {formatEth(asset.lastPrice)} ETH
-            </Row>
-            <Row display={asset.floorPrice ? 'flex' : 'none'}>FLOOR: {formatEth(asset.floorPrice)} ETH</Row>
+            {!!asset.lastPrice && (
+              <Row display={asset.lastPrice ? 'flex' : 'none'} marginRight="8">
+                LAST: {formatEth(asset.lastPrice)} ETH
```
Comment (justindomingue) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:219]: should this be in `Trans`?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 3]
```diff
@@ -214,10 +214,14 @@ const PriceTextInput = ({
...
+              </Row>
+            )}
+            {!!asset.floorPrice && (
```
Comment (justindomingue) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:222]: this wlil not render when `floorPrice===0`, is that intentional?

File: src/nft/utils/listNfts.ts [hunk 4]
```diff
@@ -62,7 +62,7 @@ const getConsiderationItems = (
   creatorFee?: ConsiderationInputItem
 } => {
   const openSeaBasisPoints = OPENSEA_DEFAULT_FEE * INVERSE_BASIS_POINTS
-  const creatorFeeBasisPoints = asset.basisPoints
+  const creatorFeeBasisPoints = asset?.basisPoints ?? 0
```
Comment (justindomingue) [inline src/nft/utils/listNfts.ts:65]: i feel like we non-null cast a lot. Should we let those fields be null until we /need/ to display them?

File: src/graphql/data/apollo.ts [hunk 5]
```diff
@@ -11,5 +12,13 @@ export const apolloClient = new ApolloClient({
     'Content-Type': 'application/json',
     Origin: 'https://app.uniswap.org',
   },
-  cache: new InMemoryCache(),
+  cache: new InMemoryCache({
+    typePolicies: {
+      Query: {
+        fields: {
+          nftBalances: relayStylePagination(),
```
Comment (thomasthachil) [inline src/graphql/data/apollo.ts:19]: do you need to pass in any `keyArgs` here for caching? Depends on if/ how you do caching, but on mobile i had issues where the entries between wallets were referencing the same query cache unless i added `ownerAddress`.

**Review by thomasthachil — COMMENTED** [review]
added context from change on mobile

File: apollo-codegen.ts (hunk 1 above)
Comment (cbachmeier) [reply apollo-codegen.ts:14]: Yes, all values can still be undefined and will require that check. This prevent redundant checking for both null and undefined

File: src/nft/components/profile/list/NFTListingsGrid.tsx (hunk 2 above)
Comment (cbachmeier) [reply src/nft/components/profile/list/NFTListingsGrid.tsx:219]: Internationalization isn't part of this migration and is handled in its own epic

File: src/nft/components/profile/list/NFTListingsGrid.tsx (hunk 3 above)
Comment (cbachmeier) [reply src/nft/components/profile/list/NFTListingsGrid.tsx:222]: yes this is intentional

File: src/nft/utils/listNfts.ts (hunk 4 above)
Comment (cbachmeier) [reply src/nft/utils/listNfts.ts:65]: This is where it's needed to calculate creator royalties, in this case if the value is undefined we can assume that no royalty amount is set

File: src/nft/components/bag/profile/utils.ts [hunk 6]
```diff
@@ -142,7 +142,7 @@ export const getListings = (sellAssets: WalletAsset[]): [CollectionRow[], Listin
   sellAssets.forEach((asset) => {
     asset.marketplaces?.forEach((marketplace: ListingMarket) => {
       const newListing = {
-        images: [asset.smallImageUrl, marketplace.icon],
+        images: [asset.smallImageUrl ?? '', marketplace.icon],
```
Comment (tinaszheng) [inline src/nft/components/bag/profile/utils.ts:145]: can you have images accept `(string | undefined)[]` instead?

File: src/nft/utils/listNfts.ts [hunk 7]
```diff
@@ -239,14 +241,14 @@ export async function signListing(
         price: parseEther(listingPrice.toString()),
         tokens: [
           {
-            token: asset.asset_contract.address,
+            token: asset.asset_contract.address ?? '',
```
Comment (tinaszheng) [inline src/nft/utils/listNfts.ts:244]: feels odd to have an empty string here be a token address 🤔 what's the expected behavior?

File: src/graphql/data/apollo.ts (hunk 5 above)
Comment (cbachmeier) [reply src/graphql/data/apollo.ts:19]: Try now, I've updated the fetch policy and able to switch wallets with no issues

File: src/nft/utils/listNfts.ts (hunk 7 above)
Comment (cbachmeier) [reply src/nft/utils/listNfts.ts]: I basically just wanted the listing to break as we shouldn't be able to list without a tokenId or collectionAddress. I've moved the check for this earlier in the function

File: src/graphql/data/nft/NftBalance.ts [hunk 8]
```diff
@@ -161,14 +127,19 @@ export function useNftBalanceQuery(
...
-    queryData
-  )
-  const walletAssets: WalletAsset[] = data.nftBalances?.edges?.map((queryAsset: NftBalanceQueryAsset) => {
-    const asset = queryAsset.node.ownedAsset
+    fetchPolicy: 'cache-and-network',
+  })
+
+  const hasNext = data?.nftBalances?.pageInfo?.hasNextPage
+  const loadMore = () =>
+    fetchMore({
```
Comment (thomasthachil) [inline src/graphql/data/nft/NftBalance.ts:135]: does this need an argument on how many to `fetch`? Believe backend has a default but might be good to explicity define the number of items in subsequent requests

Comment (cbachmeier) [reply src/graphql/data/nft/NftBalance.ts:135]: `fetchMore` will automatically use the previous params unless overwritten https://www.apollographql.com/docs/react/pagination/core-api/

**Review by justindomingue — APPROVED** [review]
migration looks okay to me

**Pull request merged**